### ISSUE TITLE
We should create an interface for Axis

### ISSUE THREAD

**Issue by olefasting** [fishfight/fishsticks#19]
I am doing some testing with WASM and I noticed that the Axis type, used as argument to `[..]::digital_value` varies from backend to backend.

We should probably create a common enum type that can be used in our API and, if needed, be converted to the appropriate type by the individual backends, in stead of using generics.

**Comment by PotatoTech** [comment]
The buttons vary between backends as well. Unfortunately, not every input in one backend has an equivalent in the other, so if I created enums for the inputs they have in common, some inputs would be unavailable. However, I can add the common set as a default feature, and if users want the backend-specific inputs, they could disable it.

**Comment by PotatoTech** [comment]
On second thought, it would be better for the common set to be a core feature that cannot be disabled. This library is meant to abstract over the backends, after all. Backend-specific inputs would then become an opt-in feature. This will be a breaking change.

**Comment by olefasting** [comment]
> On second thought, it would be better for the common set to be a core feature that cannot be disabled. This library is meant to abstract over the backends, after all. Backend-specific inputs would then become an opt-in feature. This will be a breaking change.

I agree. This seems like the best design to me, at least. We should probably bite the bullet and Break All The Things ;)
